examples/v1/taskruns/dind-sidecar.yaml

```diff
@@ -1,70 +0,0 @@
-apiVersion: tekton.dev/v1
```

Comment:
I might misunderstood but it feels like that we are converting sidecar specs to steps here https://github.com/tektoncd/pipeline/blob/a0b5cc55986818bff3c068f85e440e1236f172d1/pkg/pod/script.go#L102-L104 , which lost the *Probes Commands. Shall we use Sidecar Struct here instead?

Reply:
I'm not sure how important it is to handle these fields for converting between v1 Step and v1beta1 Step. It might be better to just drop them-- we should discuss.

The bigger problem is the code you linked to in script.go. Right now we're converting a v1beta1 sidecar to a v1beta1 step, which has no information loss. However, converting a v1 sidecar to a v1 step would have information loss. We need to update script.go to ensure these fields are still supported in sidecars, and add tests for that functionality if they don't already exist.